corgi/core/migrations/0041_custom_indexes.py

```diff
@@ -0,0 +1,31 @@
+# custom django migrations for setting pg indexes for 'latest view'
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("core", "0040_auto_20221020_1057"),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            "CREATE INDEX core_compon_latest_name_type_idx ON public.core_component \
+             USING btree (name, type) WHERE (((type)::text = 'SRPM'::text) OR \
+             (((arch)::text = 'noarch'::text) AND ((type)::text = 'CONTAINER_IMAGE'::text)) \
+             OR ((type)::text = 'RHEL_MODULE'::text))"
+        ),
+        migrations.RunSQL(
+            "CREATE INDEX core_compon_latest_type_name_idx ON public.core_component \
+             USING btree (type, name) WHERE (((type)::text = 'SRPM'::text) OR \
+             (((arch)::text = 'noarch'::text) AND ((type)::text = 'CONTAINER_IMAGE'::text)) \
+             OR ((type)::text = 'RHEL_MODULE'::text))"
+        ),
+        migrations.RunSQL(
+            "CREATE INDEX core_compon_latest_idx ON public.core_component \
+            USING btree (uuid, software_build_id, type, name, product_streams) \
+            WHERE (((type)::text = 'SRPM'::text) OR (((arch)::text = 'noarch'::text) \
+            AND ((type)::text = 'CONTAINER_IMAGE'::text)) OR ((type)::text = 'RHEL_MODULE'::text))"
+        ),
```

Comment:
Can all of these not be added on the model itself? Why are we using raw SQL here?

Reply:
these are custom partial indexes which I believe in django can kind of be represented with constraints but (for me) representing what is a perf only index in that way obscures intent... prefer to isolate perf only indexes as raw sql.

Reply:
I think these indexes are obscured if they only exist in a migration though. How is a future developer supposed to find out about them unless they want to read through 40+ migration files? At the very least I'd add a call out to this file on the affected models in models.py.

Reply:
admittedly depending on grep CREATE INDEX is not very explicit ... thx for highlighting - if it is ok with you I will add comments calling out (and will backfill comments for last custom index in the same manner )